lib/src/app/cast.js

```diff
@@ -18,58 +20,73 @@ import '../styles/player.css';
  * @param {PageLevelParameters} [pageLevelParameters]
  */
 export function init(namespace, pageLevelParameters) {
-    var mediaManager = null;
-    var castPlayer = new CastPlayer(namespace, pageLevelParameters);
-
     logger.setLevel(process.env.LOG_LEVEL);
     UIManager.setStatusCast(UIManager.status.LOADING);
 
-    // create a new instance of the Media Manager to interact with the sender app request
-    mediaManager = new cast.receiver.MediaManager(castPlayer);
-
-    // backup of the original onLoad event handler to use it after our custom logic
-    mediaManager.origOnLoad = mediaManager.onLoad;
+    const castPlayer = new CastPlayer(namespace, pageLevelParameters);
+    const mediaManager = new cast.receiver.MediaManager(castPlayer);
 
     /**
-     * onLoad assign a new implementation of the onLoad event that can sets the needed information from the
+     * Exotic class method extending by Google devs
+     * @see https://developers.google.com/cast/docs/custom_receiver (section Media)
+     * onLoad assign a new implementation of the onLoad event that can sets the
+     * needed information from the
      * incoming event
      * @param {cast.receiver.MediaManager.Event} event
      */
-    mediaManager.onLoad = function(event) {
-        logger.info('MediaManager:onLoad', event);
-        if (event && event.data && event.data.autoplay) {
-            Timer.setIdle();
-        }
-        castPlayer.setAsset(event.data.media.customData);
-        if (castPlayer.OOPlayer != null) {
-            castPlayer.notifySenders(castPlayer.OOPlayer.getState());
+    mediaManager.onLoad = (() => {
+        mediaManager.origOnLoad = mediaManager.onLoad;
+        return (event) => {
+            logger.info('MediaManager:onLoad', event);
+            const isAutoplayDesired = event && event.data && event.data.autoplay;
+            if (isAutoplayDesired) {
+                Timer.setIdle();
+            }
+            castPlayer.setAsset(event.data.media.customData);
+            if (castPlayer.OOPlayer != null) {
+                const state = castPlayer.OOPlayer.getState();
+                castPlayer.notifySenders(state);
+                if (isAutoplayDesired) {
+                    castPlayer.play();
+                }
+            }
+            mediaManager.origOnLoad(event);
         }
-        mediaManager.origOnLoad(event);
-    };
-
-    // backup of the original onPause event handler to use it after our custom logic
-    mediaManager.origPause = mediaManager.onPause;
+    })();
 
     /**
-     * onPause overrides the default behaviour to setup the default timeout when an assets it's getting paused
+     * Exotic class method extending by Google devs
+     * @see https://developers.google.com/cast/docs/custom_receiver (section Media)
+     * onPause overrides the default behaviour to setup the default timeout
+     * when an assets it's getting paused
```

Comment:
Change this last line to:
* when an asset is being paused

Reply:
+

Reply:
I dont see the change for this last line

Reply:
Sorry, fixed